src/mlpack/core/util/program_doc.hpp

```diff
@@ -18,49 +20,108 @@ namespace util {
+   * constructed.
+   *
+   * @param shortDescription A short two-sentence description of the program,
    *     what it does, and what it is useful for.
-   * @param documentation Long string containing documentation on how to use the
-   *     program and what it is.  No newline characters are necessary; this is
+   */
+  ShortDescription(const std::string& shortDescription);
+
+  /**
+   * Construct an empty ShortDescription object.
+   * (This is not meant to be used!)
+   */
+  ShortDescription();
+  std::string shortDescription;
+};
+
+class LongDescription
+{
+ public:
+  /**
+   * Construct a LongDescription object. When constructed, it will register itself
+   * with IO.  A fatal error will be thrown if more than one is constructed.
+   *
+   * @param longDescription Long string containing documentation on 
+   *     what it is.  No newline characters are necessary; this is
    *     taken care of by IO later.
-   * @param seeAlso A set of pairs of strings with useful "see also"
-   *     information; each pair is <description, url>.
    */
-  ProgramDoc(const std::string programName,
-             const std::string shortDocumentation,
-             const std::function<std::string()> documentation,
-             const std::vector<std::pair<std::string, std::string>> seeAlso);
+  LongDescription(const std::function<std::string()>& longDescription);
 
   /**
-   * Construct an empty ProgramDoc object.  (This is not meant to be used!)
+   * Construct an empty LongDescription object.  (This is not meant to be used!)
    */
-  ProgramDoc();
+  LongDescription();
+  std::function<std::string()> longDescription;
+};
 
-  //! The name of the program.
-  std::string programName;
-  //! The short documentation for the program.
-  std::string shortDocumentation;
-  //! Documentation for what the program does.
-  std::function<std::string()> documentation;
-  //! Set of see also information.
-  std::vector<std::pair<std::string, std::string>> seeAlso;
+class Example
+{
+ public:
+  /**
+   * Construct a Example object.  When constructed, it will register itself
+   * with IO.  A fatal error will be thrown if more than one is constructed.
+   *
+   * @param example Documentation on how to use the program.
+   */
+  Example(const std::function<std::string()>& example);
+
+  /**
+   * Construct an empty Example object.  (This is not meant to be used!)
+   */
+  Example();
+  std::function<std::string()> example;
 };
 
+class SeeAlso
+{
+ public:
+  /**
+   * Construct a SeeAlso object.  When constructed, it will register itself
+   * with IO.  A fatal error will be thrown if more than one is constructed.
+   *
+   * @param description Description of SeeAlso.
+   * @param link Link of SeeAlso.
+   */
+  SeeAlso(const std::string& description, const std::string& link);
+
+  /**
+   * Construct an empty SeeAlso object.  (This is not meant to be used!)
+   */
+  SeeAlso();
+  std::string description;
+  std::string link;
+};
```

Comment:
```suggestion
};

```
And one more spacing fix. :)